### PR TITLE
feat: slot change callback (provides real-time fork information)

### PR DESCRIPTION
**Pull request by sunnygleason**
PROBLEM:

* clients would like realtime slot notifications (e.g. for fork visualization)
* see solana-labs/solana#7114 for latest problem description

SOLUTION:

* hook up the websocket slot notification API

### CODE REVIEW

**Comment by sunnygleason** [comment]
see https://github.com/solana-labs/solana/pull/7114

File: src/connection.js [hunk 1]
```diff
@@ -618,6 +640,10 @@ export class Connection {
     [number]: ProgramAccountSubscriptionInfo,
   } = {};
   _programAccountChangeSubscriptionCounter: number = 0;
+  _slotSubscriptions: {
+    [number]: number,
```
Comment (CriesofCarrots) [inline src/connection.js]: Looks like this should be `[number]: SlotSubscriptionInfo`, or similar (analog of `AccountSubscriptionInfo`, will need defining.)
This should appease flow in ci...

Comment (sunnygleason) [reply src/connection.js]: (should be all set, thank you!)

**Pull request merged**